library/alloc/src/sync.rs

```diff
@@ -42,6 +41,10 @@ use crate::vec::Vec;
 #[cfg(test)]
 mod tests;
 
+fn is_dangling<T: ?Sized>(ptr: *const T) -> bool {
```

Comment:
I don't think this is a particularly good place for a helper function with such a generic name but a specific implemention

Reply:
in that case you're fine, it's not your crime :)